pvanalytics/system.py

```diff
@@ -1 +1,119 @@
 """Functions for identifying system characteristics."""
+import pandas as pd
+import numpy as np
+import pvlib
+from pvanalytics.util import _fit, _group
+
+
+def _peak_times(data):
+    minute_of_day = pd.Series(
+        data.index.hour * 60 + data.index.minute,
+        index=data.index
+    )
+    peak_minutes = _group.by_day(data).apply(
+        lambda day: pd.Timedelta(
+            minutes=_fit.quadratic_idxmax(
+                x=minute_of_day[day.index],
+                y=day,
+                model_range=range(0, 1440)
+            )
+        )
+    )
+    return pd.DatetimeIndex(
+        np.unique(data.index.date),
+        tz=data.index.tz
+    ) + peak_minutes
+
+
+def infer_orientation_least_squares(power_or_poa, daytime, tilts,
```

Comment:
I think I would prefer `_solarnoon` or `_pvfleets`, `_least_squares` is generic and could be a feature of other tilt/azimuth algorithms.